yt/frontends/enzo_e/tests/test_outputs.py

```diff
@@ -103,3 +105,27 @@ def test_critical_density():
     )
 
     assert np.abs(c1 - c2) / max(c1, c2) < 1e-3
+
+
+@requires_file(orszag_tang)
+def test_face_centered_bfields():
+    # this is based on the enzo frontend test, test_face_centered_mhdct_fields
+    ds = data_dir_load(orszag_tang)
+
+    ad = ds.all_data()
+    assert len(ds.index.grids) == 4
+
+    for field, flag in NODAL_FLAGS.items():
+        dims = ds.domain_dimensions
+        assert_equal(ad[field].shape, (dims.prod(), 2 * sum(flag)))
+
+        # the x and y domains are each split across 2 blocks. The z domain
+        # is all located on a single block
+        block_dims = (dims[0] // 2, dims[1] // 2, dims[2])
+        for grid in ds.index.grids:
+            assert_equal(grid[field].shape, block_dims + (2 * sum(flag),))
+
+    # Average of face-centered fields should be the same as cell-centered field
+    assert (ad["enzoe", "bfieldi_x"].sum(axis=-1) / 2 == ad["enzoe", "bfield_x"]).all()
+    assert (ad["enzoe", "bfieldi_y"].sum(axis=-1) / 2 == ad["enzoe", "bfield_y"]).all()
+    assert (ad["enzoe", "bfieldi_z"].sum(axis=-1) / 2 == ad["enzoe", "bfield_z"]).all()
```

Comment:
specialised array comparison methods from numpy produce much more informative messages in case of failure :)
```suggestion
    assert_array_equal(ad["enzoe", "bfieldi_x"].sum(axis=-1) / 2, ad["enzoe", "bfield_x"])
    assert_array_equal(ad["enzoe", "bfieldi_y"].sum(axis=-1) / 2, ad["enzoe", "bfield_y"])
    assert_array_equal(ad["enzoe", "bfieldi_z"].sum(axis=-1) / 2, ad["enzoe", "bfield_z"])
```